### DISCUSSION TITLE
How to detect scroll position with smooth scroll?

### DISCUSSION THREAD

**Discussion by maksderylo** [idiotWu/smooth-scrollbar#540] [FAQ] [answered]
I want to keep the benefits of smooth scroll whilst being able to perform animations depending on scroll position

**Comment by sadeghbarati** [comment, marked as answer]
Hi, 

You can use [`addListener`](https://github.com/idiotWu/smooth-scrollbar/blob/develop/docs/api.md#scrollbaraddlistener) event and store scroll position with CSS variables like 

```ts
const root = document.documentElement;
scrollbar.addListener(( { offset } ) => {
   root.style.setProperty('--scrollbar-position-y', offset.y);
});

// when you need the variable
getComputedStyle(document.documentElement).getPropertyValue('--scrollbar-position-y');
```

---

Or use [GSAP ScrollTrigger](https://greensock.com/scrolltrigger/) + [ScrollTrigger scrollerProxy](https://greensock.com/docs/v3/Plugins/ScrollTrigger/static.scrollerProxy())

https://codepen.io/GreenSock/pen/oNLqgBm

**Reply by maksderylo** [reply]
Thanks a lot I will look into it
